tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "cacerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "cacerts", Path: fmt.Sprintf("ca.%s-cert.pem", orgName)}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "signcerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "signcerts", Path: fmt.Sprintf("%s.%s-cert.pem", name, orgName)}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "keystore", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "keystore", Path: "priv_sk"}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "tlscacerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "tlscacerts", Path: fmt.Sprintf("tlsca.%s-cert.pem", orgName)}}}}}
+	volumes = append(volumes, volume)
+	if enableNodeOUs {
+		volume = v12.Volume{Name: "config", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "config", Path: "config.yaml"}}}}}
+		volumes = append(volumes, volume)
+	}
+	volume = v12.Volume{Name: "tls", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-tls", name)}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: fmt.Sprintf("%s-config", componentType), VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-config", name)}}}}
+	volumes = append(volumes, volume)
+	if dataPersistence == "true" || dataPersistence == "local" {
+		volume = v12.Volume{Name: "data-storage", VolumeSource: v12.VolumeSource{PersistentVolumeClaim: &v12.PersistentVolumeClaimVolumeSource{ClaimName: fmt.Sprintf("%s-data", name)}}}
+		volumes = append(volumes, volume)
 	}
-	return args
+	if componentType == "orderer" {
+		volume = v12.Volume{Name: "genesisblock", VolumeSource: v12.VolumeSource{Secret: &v12.SecretVolumeSource{SecretName: "genesisblock"}}}
+		volumes = append(volumes, volume)
+	}
+	return volumes
 }
 
-//GenerateConfigurationFiles - to generate all the configuration files
-func (k K8s) GenerateConfigurationFiles(upgrade bool) error {
+func (k8s K8s) generateCorePeerConfig(name, orgName, mspID string, port int32, metricsPort int32, coreConfig fabricconfig.Core) error {
```

Comment:
Can we push this up into the `fabricconfig/core.go`? Or is there something that would keep us from putting it there?

Reply:
yes agree. moved to core.go.